src/components/StringInput.tsx

```diff
@@ -1,40 +1,38 @@
 /* @jsx jsx */
 import { jsx } from '@emotion/core';
-import { Box, Flex, Input, Text } from '@stoplight/ui-kit';
+import { Box, Flex, Input } from '@stoplight/ui-kit';
 import * as React from 'react';
 
 import { IFormtronControl } from '..';
 
-import { ValidityIndicator } from './ValidityIndicator';
+import { useDiagnostics } from './hooks';
+import { Label } from './Label';
+import { Messages } from './Messages';
 
-export const StringInput: React.FunctionComponent<IFormtronControl> = ({ id, value = '', schema, onChange }) => {
-  const [validityState, changeValidityState] = React.useState<boolean | null>(null);
-
-  const onBlur = React.useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
-    changeValidityState(e.currentTarget.checkValidity());
-  }, []);
+export const StringInput: React.FunctionComponent<IFormtronControl> = ({ id, value = '', schema, onChange, path }) => {
+  const { variant } = useDiagnostics(path);
 
   return (
-    <Flex width="100%">
+    <Flex width="100%" alignItems="center">
       <Box flex="1">
-        <Text as="label" htmlFor={id}>
-          {schema.title}
-        </Text>
+        <Messages path={path}>
+          <Label htmlFor={id} variant={variant}>
+            {schema.title}
+          </Label>
+        </Messages>
       </Box>
       <Flex flex="1" width="100%">
         <Input
           type="text"
           id={id}
           value={value}
-          onChange={e => onChange(e.currentTarget.value)}
+          onChange={(e: React.SyntheticEvent<HTMLInputElement>) => onChange(e.currentTarget.value)}
           minLength={schema.minLength}
           maxLength={schema.maxLength}
           required={schema.required}
-          onBlur={onBlur}
           flex="1"
+          invalid={variant === 'invalid'}
```

Comment:
Enum!

Reply:
you don't need to make this comment on _every_ component lol

Reply:
Yeah, but I noticed that in general, it's better to comment the majority or all of them.
Otherwise, they kind of left unnoticed and I had to repeat that during the next iteration.
Not sure if that applies to you, though.